services/LARService.js

```diff
@@ -23,6 +23,47 @@ var comparePercentages = function (newPercentage, oldPercentage, threshold) {
     return true;
 };
 
+var calculateYearOnYearLoans = function (currentLoans, currentPurchaserLoans,
+            totalQuery, purchaserQuery, loanParameters, callback) {
+    var previousYearLoans,
+        previousYearPurchaserLoans;
+
+    LAR.count(totalQuery).exec()
+    .then(function (data) {
+        previousYearLoans = data;
+        return LAR.count(purchaserQuery).exec();
+    })
+    .then(function (data) {
+        previousYearPurchaserLoans = data;
+        var previousYearPercent = previousYearPurchaserLoans/previousYearLoans,
+            currentPercent = currentPurchaserLoans/currentLoans;
+
+        if (isNaN(previousYearPercent)) {
+            previousYearPercent = 0;
+        }
+        if (isNaN(currentPercent)) {
+            currentPercent = 0;
+        }
+        var result = {
+            'previousYearLoans': previousYearLoans,
+            'previousYearPurchaserLoans': previousYearPurchaserLoans,
+            'result': false
```

Comment:
Why do you not included the calculated percentages in the result? I get that you can do that calculation again in the engine or the like when constructing your error object, but why bother, when you already did them here?